kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala

```diff
@@ -102,6 +103,26 @@ class SparkProcessBuilder(
     }
   }
 
+  private lazy val sparkCoreScalaVersion: String = {
```

Comment:
we can make it testable by breaking into a `function` and a `lazy val`

```
  private[kyuubi] def extractScalaVersion(sparkJars: Seq[String]): String = {
    val r = """^spark-core_(\d\.\d+).*.jar$""".r
    sparkJars.collectFirst { case r(scalaVersion) => scalaVersion }.getOrElse {
      throw new KyuubiException("xxxx")
    }
  }

  private lazy val sparkCoreScalaVersion: String = {
    val sparkJars = Files.list(Paths.get(sparkHome, "jars"))
      .iterator().asScala.map(_.getFileName.toString).toSeq
    extractScalaVersion(sparkJars)
  }
```

Reply:
Thanks. Broken into functions with unit test added.